pulsar-client-cpp/CMakeLists.txt

```diff
@@ -195,9 +195,13 @@ MESSAGE(STATUS "HAS_SNAPPY: ${HAS_SNAPPY}")
 set(ADDITIONAL_LIBRARIES $ENV{PULSAR_ADDITIONAL_LIBRARIES})
 link_directories( $ENV{PULSAR_ADDITIONAL_LIBRARY_PATH} )
 
+set(AUTOGEN_DIR ${CMAKE_BINARY_DIR}/generated)
```

Comment:
Should we also add `generated` dir to the `.gitignore`?

Reply:
@merlimat not necessary for now as files would be ignored by `pd.{cc,h}` suffix, it would be great to add `generated` directory, and the suffix pattern could be removed, i'll add it. Actually i always new a `build| directory for cmake, would we suggest to do like that and add `build` directory to .gitignore also?

Reply:
@merlimat `gernerated` added to `.gitignore`, PTAL.